src/qt/walletmodel.cpp

```diff
@@ -96,6 +102,38 @@ void WalletModel::updateStatus()
         emit encryptionStatusChanged(newEncryptionStatus);
 }
 
+void WalletModel::forceUpdateBalances()
+{
+    // boost::timer t;
+    // Get required locks upfront. This avoids the GUI from getting stuck on
+    // periodical polls if the core is holding the locks for a longer time -
+    // for example, during a wallet rescan.
+    TRY_LOCK(cs_main, lockMain);
```

Comment:
I'm not 100 % sure about the scope of the locks, but as far as I can see then `WalletModel::pollBalanceChanged()` is the only place where `WalletModel::forceUpdateBalances()` is called, and locks are already set there, see: [walletmodel.cpp#L142-L147](https://github.com/zathras-crypto/mastercore/blob/0.0.9-Z-UIAlpha/src/qt/walletmodel.cpp#L142-L147)

So I assume, it will never be possible to lock here a second time?